### DISCUSSION TITLE
prometheus API export data too large

### DISCUSSION THREAD

**Discussion by warrenincn** [apache/apisix#7513] [General]
apisix version: master
environment：K8S
scenario：Use apisix_bandwidth metrics to accomplish upstream's auto scale. Upstream's nodes use K8S pod ip. The pod ip changes  a lot, lead to the prometheus export data too large(one route enable prometheus plugin, run 2 weeks, the export data now is 13839lines/1.7MB).
Relevant code：
https://github.com/apache/apisix/blob/44d20c391767d3d42c5111ea7ef6318e84633996/apisix/plugins/prometheus/exporter.lua#L229
In our scenario, the label: node(balancer_ip) is useless.
Any idea to settle this?

**Comment by tzssangglass** [comment]
ref: https://github.com/apache/apisix/issues/4273

we are talking about that add some configuration items of the Prometheus plugin to support selecting what data to collect

**Reply by warrenincn** [reply]
How about put some label's value into fixed mock value according to route prometheus plugin's configuration? Some thing like below:
```
{
    "prefer_name":true,
    "access":[
        {
            "metric":"server_picker",
            "mock_label":[

            ]
        }
    ],
    "log":[
        {
            "metric":"bandwidth",
            "mock_label":[
                "node"
            ]
        }
    ]
}
```

**Reply by tzssangglass** [reply]
I think it would be more direct to let the user specify metrics
